app/src/main/java/com/example/quizacademytask/StackListAdapter.kt

```diff
@@ -74,27 +76,38 @@ class StackListAdapter(
     }
 
     @SuppressLint("ResourceAsColor")
-    fun paintSelected(view: View) {
-        view.findViewById<TextView>(R.id.rowText).setTextColor(0xFF87cefa.toInt())
+    fun paintSelected(view: TextView) {
+        view.setTextColor(0xFF87cefa.toInt())
+    }
+
+    @SuppressLint("ResourceAsColor")
+    fun paintUnselected(view: TextView) {
+        view.setTextColor(0xFFFDFDFD.toInt())
     }
 
     fun emptyList() {
+        val x = recyclerView.children.count()
         recyclerView.children.forEach {
             paintUnselected(it.findViewById(R.id.rowText))
         }
         selectedItems.clear()
     }
 
-    @SuppressLint("ResourceAsColor")
-    fun paintUnselected(view: TextView) {
-        view.setTextColor(0xFFFDFDFD.toInt())
+    fun restoreSelectedPaint() {
```

Comment:
I'm having difficulties re-coloring the selected items on rotate. Persisting selectedItems works however iterating over recyclerView to re-color the rows doesn't since it seems to be not inflated yet.  I've tried a bunch but nothing properly worked. What would the best approach be here?

Reply:
your recycler view adapter should get positions of selected items and use them in onBindViewHolder to highlight the items